internal/storage/download.go

```diff
@@ -39,8 +38,7 @@ const prefix = "_temporary-"
 var secureCiphers = supportedCipherSuites()
 
 // downloadArtifact tries to resume previous download operation or perform a new download.
-func downloadArtifact(to string, artifact *Artifact, progress progressBytes, serverCert string, retryCount int, retryInterval time.Duration,
-	done chan struct{}) error {
+func downloadArtifact(to string, artifact *Artifact, progress progressBytes, serverCert string, retryCount int, retryInterval time.Duration, postProcess func(fileName string) error, done chan struct{}) error {
```

Comment:
There is no line length limit in Go, but it is unpleasant to scroll. Separate in two lines and indent. Also a type for the `postProcess` function could be defined, as it could be used here and in `storage.go`.